contrib/epee/include/net/abstract_tcp_server2.inl

```diff
@@ -1003,8 +1003,23 @@ POP_WARNINGS
     }else
     {
       _erro("Some problems at accept: " << e.message() << ", connections_count = " << m_sock_count);
+      misc_utils::sleep_no_w(100);
+      new_connection_.reset(new connection<t_protocol_handler>(io_service_, m_config, m_sock_count, m_sock_number, m_pfilter, m_connection_type));
```

Comment:
Since we now do this reset/accept in all three of these if/else/catch blocks, would it make sense to de-dupe those and just do it at the end of the function, once?

Reply:
It *seems* it might be able to be moved from the main code after the start call etc. Not sure whether that call can take a while in specal cases, which would make the server reject connections while that's going on. The two ones on the error paths definitely can be though.

Reply:
Cool, up to you. Probably makes sense to leave the one in the main loop as-is to avoid creating any new potential issues.

Reply:
I'd completely forgot about this. Fixed.